utils/firestore.js

```diff
@@ -1237,6 +1237,100 @@ const searchSpecimen = async (masterSpecimenId, siteCode, allSitesFlag) => {
+        else {
+            let shipBoxObj = shipBoxMatch[0];
+            const collectionIdArray = [];
+            const bagConceptIdList = Object.values(fieldMapping.bagContainerCids);
+
+            for (let key in shipBoxObj) {
+            // check if key is in bagConceptIdList array of conceptIds
+                if (bagConceptIdList.includes(parseInt(key))) {
+                    const bagContainerContent = shipBoxObj[key]; 
+                    const bloodUrineScan = fieldMapping.tubesBagsCids.biohazardBagScan;
+                    const mouthwashScan = fieldMapping.tubesBagsCids.biohazardMouthwashBagScan;
+                    const orphanScan = fieldMapping.tubesBagsCids.biohazardMouthwashBagScan;
+
+                    // Loop through the bagContainerContent to find the collectionId
+                    for (let key in bagContainerContent) {
+                        if (parseInt(key) === bloodUrineScan || 
+                            parseInt(key) === mouthwashScan || 
+                            parseInt(key) === orphanScan) {
+                            if (bagContainerContent[key] !== '') {
+                                // extract the collectionId and push to collectionIdArray
+                                const collectionIdString = bagContainerContent[key].split(" ")[0]
+                                // check if collectionIdString is already in, if it isn't push it
+                                if (!collectionIdArray.includes(collectionIdString)) {
+                                    collectionIdArray.push(collectionIdString);
+                                }
+                                break;
+                            } 
+                        }   
+                    }
+                }
+            }
+            return collectionIdArray
+        } 
+    } catch (error) {
+        throw new Error("getBiospecimenCollectionIdsFromBox() error.", {cause: error});
+    }
+}
+/** 
+ * return an array of biospecimen documents that match the healthcare provider and collectionId from collectionIdArray
+ * calls getBiospecimenCollectionIdsFromBox to get the collectionIdArray
+ * query the biospecimen collection for documents that match the healthcare provider and collectionIds in collectionIdArray
+ * @param {number} requestedSite - site code of the site 
+ * @param {string} boxId - boxId of the box
+*/
+const searchSpecimenBySiteAndBoxId = async (requestedSite, boxId) => {
+    try {
+        const collectionIdArray = await getBiospecimenCollectionIdsFromBox(requestedSite, boxId);
+        const snapshot = await db.collection('biospecimen')
+                                .where(fieldMapping.healthCareProvider.toString(), "==", requestedSite)
+                                .where(fieldMapping.collectionId.toString(), "in", collectionIdArray).get();
+        const biospecimenDocs = [];
+
+        if (snapshot.empty) return;
+        else {
```

Comment:
nit: since you're returning if (snapshot.empty) the else {} isn't needed. You can just execute the for loop without wrapping it in the else {} clause